### PR TITLE
various undefined fixes from sentry

### PR DESCRIPTION
**Pull request by delphiactual**

### CODE REVIEW

File: src/app/dim-api/selectors.ts [hunk 1]
```diff
@@ -22,4 +22,4 @@ export const currentProfileSelector = createSelector(
  * TODO: Sort/trim this list
  */
 export const recentSearchesSelector = (state: RootState) =>
-  state.dimApi.searches[destinyVersionSelector(state)];
+  state.dimApi.searches?.[destinyVersionSelector(state)];
```
Comment (delphiactual) [inline src/app/dim-api/selectors.ts:25]: https://sentry.io/share/issue/a18b92493cde444db4eeb77af7dab091/

File: src/app/inventory/d2-stores.ts [hunk 2]
```diff
@@ -297,7 +297,7 @@ function makeD2StoresService(): D2StoreServiceType {
     const itemComponents = profileInfo.itemComponents;
     const progressions = profileInfo.characterProgressions.data?.[characterId]?.progressions || [];
     const uninstancedItemObjectives =
-      profileInfo.characterProgressions.data?.[characterId].uninstancedItemObjectives || [];
+      profileInfo.characterProgressions.data?.[characterId]?.uninstancedItemObjectives || [];
```
Comment (delphiactual) [inline src/app/inventory/d2-stores.ts:300]: https://sentry.io/share/issue/ac6de6b858b74b5cbda1ed4265c5720f/

File: src/app/loadout-builder/filter/FilterBuilds.tsx [hunk 3]
```diff
@@ -77,7 +77,7 @@ export default function FilterBuilds({
           </label>
           <RangeSelector
             min={750}
-            max={parseInt(selectedStore.stats.maxGearPower!.value.toString(), 10)}
+            max={parseInt(selectedStore.stats.maxGearPower?.value.toString() ?? '750', 10)}
```
Comment (delphiactual) [inline src/app/loadout-builder/filter/FilterBuilds.tsx:80]: https://sentry.io/share/issue/639a0aa388a94fb2861bad11be1b7216/

File: src/app/search/autocomplete.ts [hunk 4]
```diff
@@ -145,7 +145,9 @@ export function filterSortRecentSearches(query: string, recentSearches: Search[]
   //       we aren't going to do different sorts for query vs. non-query
   const recentSearchesForQuery = query
     ? recentSearches.filter((s) => s.query.includes(query))
-    : Array.from(recentSearches);
+    : recentSearches
+    ? Array.from(recentSearches)
+    : [];
```
Comment (delphiactual) [inline src/app/search/autocomplete.ts:150]: https://sentry.io/share/issue/fae80d19aa9f437793d8208932d5e079/

File: src/app/dim-ui/CustomStatTotal.tsx [hunk 5]
```diff
@@ -59,7 +59,7 @@ export function StatTotalToggle({
                 .map((statHash) => (
                   <StatToggleButton
                     key={statHash}
-                    stat={defs.Stat.get(statHash)}
+                    stat={statHash && defs.Stat.get(statHash)}
```
Comment (delphiactual) [inline src/app/dim-ui/CustomStatTotal.tsx:62]: https://sentry.io/share/issue/5e9935174ac744ffa6a8ebdc3084adc7/

This does not seem to be the right fix, as it is `def` that is somehow undefined

Comment (nev-r) [reply src/app/dim-ui/CustomStatTotal.tsx:62]: this is a known race condition where the settings page can i think load before the manifest is loaded, and therefore the stat icons which rely on the manifest, fail to look up

File: src/app/dim-api/selectors.ts (hunk 1 above)
Comment (bhollis) [reply src/app/dim-api/selectors.ts:25]: This shouldn't be necessary - I think I fixed it higher up by making sure that searches is always defined.

File: src/app/loadout-builder/filter/FilterBuilds.tsx (hunk 3 above)
Comment (bhollis) [reply src/app/loadout-builder/filter/FilterBuilds.tsx:80]: This is super weird, how can a store not have maxGearPower?

File: src/app/search/autocomplete.ts (hunk 4 above)
Comment (bhollis) [reply src/app/search/autocomplete.ts:150]: This should always be defined now as well

File: src/app/dim-ui/CustomStatTotal.tsx (hunk 5 above)
Comment (bhollis) [reply src/app/dim-ui/CustomStatTotal.tsx:62]: We should just block the search page on loading defs

**Pull request merged**